RestartManager.js

```diff
@@ -26,6 +28,12 @@ const RestartManager = (() => {
 
     function restartApp(onlyIfUpdateIsPending = false) {
         if (_allowed) {
+            if (_inProgress) {
+                log("A restart request is already in progress or queued");
+                return;
+            }
+            // The restart won't execute if `onlyIfUpdateIsPending === true` and there is no pending update.
+            _inProgress = !onlyIfUpdateIsPending || !!(NativeCodePush.getUpdateMetadata(CodePush.UpdateState.PENDING));
```

Comment:
yea, talked with @geof90 about it offline, seems like the sync in progress approach is the best way to go